object.js

```diff
@@ -0,0 +1,209 @@
+// ----------------------------------------------------
+// Dice - Beat That! V3 (Keep Score, Variable No. of Players, Leaderboard)
+// ----------------------------------------------------
+
+// Game Modes
+const CHOOSE_PLAYERS_MODE = 'CHOOSE_PLAYERS_MODE';
+const CHOOSE_DICENUMBER_MODE = 'CHOOSE_DICENUMBER_MODE';
+const ORDERING_MODE = 'ORDERING_MODE';
+const SCORING_MODE = 'SCORING_MODE';
+
+// Start game with dice rolling mode
+var mode = CHOOSE_PLAYERS_MODE;
+// Store every Player in array & all their scores
+var players = [];
+var allPlayerScoreArray = [];
+// Keep track of number of dice input at the start & current game round
+var noOfDice;
+var trackPlayer = 0;
+
+/**
+ * Get dice rolls for curr player and populate current player's dice array
+ * Return the new dice rolls
+ */
+var rollDice = function () {
+  var randInt = Math.floor(Math.random() * 6) + 1;
+  return randInt;
+};
+
+/** Create Object to store game details of each player */
+class Player {
+  /**
+  * Tracks the initial dice roll, reordered dice & score
+  * @param {Array<Number>} initialDiceArray original dice number
+  * @param {Number} orderedDice store reordered dice number
+  * @param {Array<Number>} scoreArray store series of orderedDice for scoring
```

Comment:
Great initiative! This usage of JSDoc commenting isn't exact. in particular, @params refer to function parameters, but in this case there is only 1 function parameter `orderedDice`. In this case, better to add inline comment above each of lines 38-40